repair/repair.cc

```diff
@@ -640,53 +640,61 @@ future<> shard_repair_task_impl::repair_range(const dht::token_range& range, ::t
     auto mandatory_neighbors = std::move(r_neighbors.mandatory);
     auto live_neighbors = boost::copy_range<std::vector<gms::inet_address>>(neighbors |
                 boost::adaptors::filtered([this] (const gms::inet_address& node) { return gossiper.is_alive(node); }));
-    for (auto& node : mandatory_neighbors) {
-        auto it = std::find(live_neighbors.begin(), live_neighbors.end(), node);
-        if (it == live_neighbors.end()) {
-            nr_failed_ranges++;
+
+    auto task_impl_ptr = std::make_unique<range_repair_task_impl>(rs.get_repair_module().shared_from_this(), tasks::task_id::create_random_id(), _status.keyspace, table_name, format("{}", reason), get_repair_uniq_id(), *this, range, table_id, std::move(neighbors), std::move(mandatory_neighbors), std::move(live_neighbors), ranges_index);
+    auto task = co_await rs.get_repair_module().make_task(std::move(task_impl_ptr), get_repair_uniq_id().task_info);
+    task->start();
+    co_await task->done();
+}
+
+future<> range_repair_task_impl::run() {
+    for (auto& node : _mandatory_neighbors) {
+        auto it = std::find(_live_neighbors.begin(), _live_neighbors.end(), node);
+        if (it == _live_neighbors.end()) {
+            _shard_task.nr_failed_ranges++;
             auto status = format("failed: mandatory neighbor={} is not alive", node);
             rlogger.error("repair[{}]: Repair {} out of {} ranges, shard={}, keyspace={}, table={}, range={}, peers={}, live_peers={}, status={}",
-                    id.uuid(), ranges_index, ranges_size(), id.shard(), _status.keyspace, table_names(), range, neighbors, live_neighbors, status);
-            abort_repair_info();
+                    _shard_task.id.uuid(), _ranges_index, _shard_task.ranges_size(), _shard_task.id.shard(), _status.keyspace, _shard_task.table_names(), _range, _neighbors, _live_neighbors, status);
```

Comment:
it seems like a preexisting bug that we print `table_names()` rather than the particular table the task is working on, no?

Reply:
cc @asias

Reply:
Yes, after 3dc9a81d02a26d021ad1954d57b574ad4899561d, we now only repair one table per range instead of all tables per range.

Reply:
Yes, since 3dc9a81d02a26d021ad1954d57b574ad4899561d, it repairs one table per range instead of all tables per range.